### PR TITLE
#1 HomePage added & some base components

### PR DESCRIPTION
**Pull request by ivanhuay**
Homepage added & some base components #1 

also, this is related to https://github.com/open-challenge/open-challenge-api/pull/7 so if you want to manually test this PR please use `feat/v0.0.1` on the API.

### CODE REVIEW

File: src/components/code-editor/CodeEditor.js [hunk 1]
```diff
@@ -49,7 +49,8 @@ const CodeEditor = ({ id }) => {
     getChallenge(id).then(challenge => {
       updateCode(challenge.code);
       setCurrentCode(challenge.code);
-      setDefaultArgs(challenge.datasets[0].dataset);
+      const datasets = challenge.datasets;
+      setDefaultArgs(datasets.lenght > 0 ? datasets[0].dataset : null);
```
Comment (HernanGH) [inline src/components/code-editor/CodeEditor.js]: Please replace `datasets.lenght` for `datasets.length`

File: src/components/home-card/index.js [hunk 2]
```diff
@@ -0,0 +1,21 @@
+import React from 'react';
+import { useHistory } from 'react-router-dom';
+import { Card, Button } from '../base';
+import { newChallenge } from '../../services/challenge';
+const HomeCard = () => {
+  const history = useHistory();
+  const handleClick = () => {
+    newChallenge().then(resp => {
```
Comment (HernanGH) [inline src/components/home-card/index.js]: Just suggestion 
```
const handleClick = () => newChallenge()
    .then(resp => history.push(`/challenge/${resp.auth.edition}`));
```

File: src/components/code-editor/CodeEditor.js (hunk 1 above)
Comment (ivanhuay) [reply src/components/code-editor/CodeEditor.js]: nice catch! 😄

File: src/components/home-card/index.js (hunk 2 above)
Comment (ivanhuay) [reply src/components/home-card/index.js]: the current `eslint/prettier` configuration force me to do it in that way but we can change this config later 🤔

**Comment by ivanhuay** [comment]
@HernanGH updated

**Pull request merged**